lisa/transformers/kernel_source_installer.py

```diff
@@ -114,22 +120,59 @@ def install(self) -> str:
         # modify code
         self._modify_code(node=node, code_path=code_path)
 
-        self._build_code(node=node, code_path=code_path)
+        config_type = source.get_config_type()
+        self._build_code(node=node, code_path=code_path, config_type=config_type)
 
         self._install_build(node=node, code_path=code_path)
 
-        result = node.execute(
-            "make kernelrelease 2>/dev/null", cwd=code_path, shell=True
-        )
+        kernel_version = ""
+        if self.__is_dom0(node) and config_type == "dom0":
+            # If it is dom0,
+            # Name of the current kernel should be vmlinuz-<kernel version>
+            uname = node.tools[Uname]
+            current_kernel = uname.get_linux_information().kernel_version_raw
+            current_kernel_binary = f"vmlinuz-{current_kernel}"
+
+            # Copy the binary to /boot/efi from : arch/x86/boot/bzImage
+            source_path = code_path.joinpath("arch/x86/boot/bzImage")
+            new_kernel_binary = "vmlinuz-%s" % str(uuid.uuid4())
+            destination_path = f"/boot/efi/{new_kernel_binary}"
+            cp = node.tools[Cp]
+            cp.copy(
+                src=source_path,
+                dest=PurePath(destination_path),
+                sudo=True,
+            )
 
-        kernel_version = result.stdout
-        result.assert_exit_code(0, f"failed on get kernel version: {kernel_version}")
+            # Modify the linuxloader.conf to point new kernel binary
+            ll_conf_file = "/boot/efi/linuxloader.conf"
+            sed = self._node.tools[Sed]
+            sed.substitute(
+                regexp=current_kernel_binary,
```

Comment:
Don't depend on the old value. Just find `VMLINUZ=` in the file and replace the value.

Reply:
There is no VMLINUZ= in the line where we need to change. It is **KERNEL_PATH=**

I will add that snipped in the string of regex and replaced one

Reply:
Added it